tests/integration/Owncloud/OcisPhpSdk/DriveTest.php

```diff
@@ -343,4 +344,264 @@ public function testSetExpirationDate(): void
+            );
+            $this->assertSame(
+                $this->drive->getName(),
+                $receivedInvitationDrive->getName(),
+                "Expected shared drive name to be " . $this->drive->getName() . " but found " . $receivedInvitationDrive->getName()
+            );
+            $this->assertSame(
+                'myfolder',
+                $receivedInvitationDrive->getResources()[0]->getName(),
+                "Expected resource name to be myfolder"
+                . " but found " . $receivedInvitationDrive->getResources()[0]->getName()
+            );
+        } catch(EndPointNotImplementedException) {
+            // test should fail if ocis version is less than 6.0.0
+            $this->fail("EndPointNotImplementedException was thrown unexpectedly");
+        }
+    }
+
+    /**
+     * @throws ForbiddenException
+     * @throws InvalidResponseException
+     * @throws BadRequestException
+     * @throws EndPointNotImplementedException
+     * @throws UnauthorizedException
+     * @throws HttpException
+     * @throws NotFoundException
+     * @throws InternalServerErrorException
+     * @throws \Exception
+     */
+    public function testGetMyDriveForDriveShareReceiver(): void
+    {
+        $katherineOcis = $this->getOcis('katherine', 'gemini');
+        $katherineDrive = $katherineOcis->createDrive('katherine Project Drive');
+        $this->createdDrives[] = $katherineDrive->getId();
+        $katherine = $this->ocis->getUsers('katherine')[0];
+
+        $managerRole = null;
+
+        if (getenv('OCIS_VERSION') === "stable") {
+            // ocis version < 6.0.0 doesn't support root endpoint so mocking getRoles values
+            $role = [
+                "id" => "312c0871-5ef7-4b3a-85b6-0e4074c64049",
+                "description" => "Allows managing a space",
+                "displayName" => "Manager",
+                "@libre.graph.weight" => 3
+            ];
+
+            $shareRoles = [new SharingRole(new UnifiedRoleDefinition($role))];
+        } else {
+            $shareRoles = $this->drive->getRoles();
+        }
+        foreach ($shareRoles as $role) {
+            if ($role->getId() === self::getPermissionsRoleIdByName('Manager')) {
+                $managerRole = $role;
+                break;
+            }
+        }
+
+        if (empty($managerRole)) {
+            throw new \Error(
+                "manager role not found "
+            );
+        }
+        // ocis stable doesn't support root endpoint
+        if (getenv('OCIS_VERSION') === "stable") {
+            $this->expectException(EndPointNotImplementedException::class);
+            $this->expectExceptionMessage("This method is not implemented in this ocis version");
+            $this->drive->invite($katherine, $managerRole);
+        }
+
+        try {
+            $driveInvitation = $this->drive->invite($katherine, $managerRole);
+
+            $this->assertInstanceOf(
```

Comment:
no need to test the same thing over and over again.
let every test test one specific thing